packages/toolkit/src/query/tests/createApi.test.ts

```diff
@@ -522,6 +531,71 @@ describe('endpoint definition typings', () => {
         ['modified2', { ...commonBaseQueryApi, forced: undefined }, undefined],
       ])
     })
+
+    test('updated transform response types', async () => {
+      const baseApi = createApi({
+        baseQuery: fetchBaseQuery({ baseUrl: 'https://example.com' }),
+        tagTypes: ['old'],
+        endpoints: (build) => ({
+          query1: build.query<'out1', void>({ query: () => 'success' }),
+          mutation1: build.mutation<'out1', void>({ query: () => 'success' }),
+        }),
+      })
+
+      type Transformed = { value: string }
+
+      type Definitions = DefinitionsFromApi<typeof api>
+      type TagTypes = TagTypesFromApi<typeof api>
+
+      type Q1Definition = OverrideResultType<Definitions['query1'], Transformed>
+      type M1Definition = OverrideResultType<
+        Definitions['mutation1'],
+        Transformed
+      >
+
+      type UpdatedDefitions = Omit<Definitions, 'query1' | 'mutation1'> & {
+        query1: Q1Definition
+        mutation1: M1Definition
+      }
+
+      const enhancedApi = baseApi.enhanceEndpoints<TagTypes, UpdatedDefitions>({
```

Comment:
i don't love that these generics are being provided manually, rather than being inferred as they would be in actual usage - could the test be reworked to test the inferrence?

Reply:
The manual generics are the purpose of this pr. Code generation forces the output of `transformResponse` to match its input since at generation time the output type of a  `transformResponse` function (which will be supplied later) is not known. The purpose of pr is to enable manual override for `transformResponse` return type so that it can be used inside of `enhanceEndpoints`. If inference was possible, this pr would not be needed